lib/object-path-immutable-transaction.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+const _assign = require('lodash.assign');
+const objectPath = require('object-path');
+const _isArray = require('lodash.isarray');
+const _isPlainObject = require('lodash.isplainobject');
+
+function ensureCloned(path, source, dest) {
+	if (!_isArray(path)) {
+		path = ('' + path).split('.');
+	}
+
+	// ensure all ancestor path segments are cloned
+	for (let i = 0; i < path.length; i++) {
+		if (dest[path[i]] === source[path[i]]) {
+			if (_isArray(source[path[i]])) {
+				dest[path[i]] = source[path[i]].slice();
+			} else {
```

Comment:
I thought about this but we don't need that check for Redux stores. I'm going to wait and see what the object-path-immutable folks want. If they say no to the patch, we can think about throwing in the case that any ancestor is not a plain object or array.